### PR TITLE
remove fields deleted in PluginConfig and PluginVersionMapping

### PR DESCRIPTION
**Pull request by cyinma**
These fields were deprecated and are now safe to be removed.

### CODE REVIEW

File: proto/buf/alpha/registry/v1alpha1/plugin.proto [hunk 1]
```diff
@@ -83,9 +83,6 @@ message PluginConfig {
   // Parameters that should be provided to the plugin. These are
   // joined with a "," before being provided to the plugin at runtime.
   repeated string parameters = 3;
-  // True if the source plugin has been deleted. This dooms
-  // the template.
-  bool deleted = 4 [deprecated=true];
```
Comment (mfridman) [inline proto/buf/alpha/registry/v1alpha1/plugin.proto:88]: As a best practice, we should either use `reserved` or decrement the field numbers. Leaving the `4` omitted will cause confusing to the next person.

Comment (mfridman) [reply proto/buf/alpha/registry/v1alpha1/plugin.proto:88]: @cyinma When you get a chance, can you update the Buf style guide ?

Framing the question: "How do I remove a field from our public Protobuf definitions?"

**Review by mfridman — APPROVED** [review]

File: proto/buf/alpha/registry/v1alpha1/plugin.proto [hunk 2]
```diff
@@ -76,16 +76,14 @@ message Template {
 
 // PluginConfig defines a runtime configuration for a plugin.
 message PluginConfig {
+  reserved 4;
```
Comment (johanbrandhorst) [inline proto/buf/alpha/registry/v1alpha1/plugin.proto:79]: We need to reserve the name too
```suggestion
  reserved 4;
  reserved "deleted";
```

File: proto/buf/alpha/registry/v1alpha1/plugin.proto [hunk 3]
```diff
@@ -110,15 +108,13 @@ message TemplateVersion {
 
 // PluginVersionMapping maps a plugin_id to a version.
 message PluginVersionMapping {
+  reserved 4;
```
Comment (johanbrandhorst) [inline proto/buf/alpha/registry/v1alpha1/plugin.proto:111]: Same here

**Pull request merged**